lib/ansible/module_utils/facts/network/iscsi.py

```diff
@@ -80,23 +80,32 @@ def collect(self, module=None, collected_facts=None):
                     break
         elif sys.platform.startswith('aix'):
             if module is not None:
-                rc, out, err = module.run_command('/usr/sbin/lsattr -E -l iscsi0 | grep initiator_name')
+                cmd = module.get_bin_path('lsattr', required=True)
+                cmd = cmd + " -E -l iscsi0"
+                rc, out, err = module.run_command(cmd)
                 if out:
-                    iscsi_facts['iscsi_iqn'] = out.split()[1].rstrip()
+                    for line in out.splitlines():
+                        if 'initiator_name' in line:
+                            iscsi_facts['iscsi_iqn'] = line.split()[1].rstrip()
             else:
-                aixcmd = '/usr/sbin/lsattr -E -l iscsi0 | grep initiator_name'
+                aixcmd = '/usr/sbin/lsattr -E -l iscsi0'
                 aixret = subprocess.check_output(aixcmd, shell=True)
                 if aixret[0].isalpha():
-                    iscsi_facts['iscsi_iqn'] = aixret.split()[1].rstrip()
+                    for line in aixret.splitlines():
```

Comment:
might want to make common function instead of creating the same for loop several times?

Reply:
done